dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowVersionDAO.java

```diff
@@ -55,4 +93,45 @@ public List<WorkflowVersion> getTagsByWorkflowIdOrderedByLastModified(long workf
         query.setMaxResults(limit);
         return query.getResultList();
     }
+
+    private List<Predicate> processQuery(String sortCol, String sortOrder, CriteriaBuilder cb, CriteriaQuery query, Root<WorkflowVersion> version) {
+        List<Predicate> predicates = new ArrayList<>();
+
+        if (!Strings.isNullOrEmpty(sortCol)) {
+            if ("open".equalsIgnoreCase(sortCol)) {
+                if ("desc".equalsIgnoreCase(sortOrder)) {
+                    query.orderBy(cb.desc(version.get("versionMetadata").get("publicAccessibleTestParameterFile")), cb.desc(version.get("id")));
+                } else {
+                    query.orderBy(cb.asc(version.get("versionMetadata").get("publicAccessibleTestParameterFile")), cb.asc(version.get("id")));
+                }
+            } else if ("descriptorTypeVersions".equals(sortCol)) {
+                if ("desc".equalsIgnoreCase(sortOrder)) {
+                    query.orderBy(cb.desc(version.get("versionMetadata").get("descriptorTypeVersions")), cb.desc(version.get("id")));
+                } else {
+                    query.orderBy(cb.asc(version.get("versionMetadata").get("descriptorTypeVersions")), cb.asc(version.get("id")));
+                }
+            } else {
+                boolean hasSortCol = version.getModel()
+                        .getAttributes()
+                        .stream()
+                        .map(Attribute::getName)
+                        .anyMatch(sortCol::equals);
+
+                if (!hasSortCol) {
+                    LOG.error(INVALID_SORTCOL_MESSAGE);
+                    throw new CustomWebApplicationException(INVALID_SORTCOL_MESSAGE,
+                            HttpStatus.SC_BAD_REQUEST);
+
+                } else {
+                    Path<Object> sortPath = version.get(sortCol);
+                    if ("asc".equalsIgnoreCase(sortOrder)) {
+                        query.orderBy(cb.asc(sortPath), cb.asc(version.get("id")));
+                    } else {
+                        query.orderBy(cb.desc(sortPath), cb.desc(version.get("id")));
+                    }
+                }
```

Comment:
Would recommend trying to streamline this method to reduce the amount of duplicate/similar code.  One way would be to calculate a sort path from the various `sortCol` values, then apply the sort path and order to the query at the end.  Something like:
```
    private List<Predicate> processQuery(String sortCol, String sortOrder, CriteriaBuilder cb, CriteriaQuery query, Root<WorkflowVersion> version) {
        List<Predicate> predicates = new ArrayList<>();

        if (!Strings.isNullOrEmpty(sortCol)) {
            Path<Object> sortPath;
            if ("open".equalsIgnoreCase(sortCol)) {
                sortPath = version.get("versionMetadata").get("publicAccessibleTestParameterFile");
            } else if ("descriptorTypeVersions".equals(sortCol)) {
                sortPath = version.get("versionMetadata").get("descriptorTypeVersions");
            } else {
                boolean hasSortCol = version.getModel()
                        .getAttributes()
                        .stream()
                        .map(Attribute::getName)
                        .anyMatch(sortCol::equals);

                if (!hasSortCol) {
                    LOG.error(INVALID_SORTCOL_MESSAGE);
                    throw new CustomWebApplicationException(INVALID_SORTCOL_MESSAGE,
                            HttpStatus.SC_BAD_REQUEST);

                }
                sortPath = version.get(sortCol);
            }
            if ("asc".equalsIgnoreCase(sortOrder)) {
                query.orderBy(cb.asc(sortPath), cb.asc(version.get("id")));
            } else {
                query.orderBy(cb.desc(sortPath), cb.desc(version.get("id")));
            }
        }
        return predicates;
    }
}
```

Reply:
Done! Thank you